flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/relational/connection/JdbcConnectionPools.java

```diff
@@ -83,4 +84,14 @@ public String getJdbcUrl(
         }
         return jdbcConnectionPoolFactory.getJdbcUrl(sourceConfig);
     }
+
+    public void clear() throws IOException {
+        if (instance != null) {
```

Comment:
We will use the `instance` to invoke this method and the `instance` will not be null. We could delete this judgement.